tiledb/sm/rest/rest_client.cc

```diff
@@ -286,19 +281,16 @@ Status RestClient::post_query_submit(
       std::placeholders::_4,
       &scratch,
       query,
-      copy_state,
-      &user_buffers_overflowed);
+      copy_state);
 
   const Status st = curlc.post_data(
       url, serialization_type_, &serialized, std::move(write_cb));
 
-  // When 'user_buffers_overflowed' is true, we will return all of
-  // the known query data even if the response parsing was unsuccessful.
-  // This is a temporary work-around to allow returning an incomplete
-  // query when the client-side buffer is too small to completely
-  // store the returned query.
-  if (!user_buffers_overflowed) {
-    RETURN_NOT_OK(st);
+  // We can safely return 'query' to the client even when the request
+  // fails, because we garauntee that 'query' will be in a good state,
+  // even if we were unable to deserialize all query objects.
+  if (!st.ok()) {
+    LOG_STATUS(st);
```

Comment:
Is it a good idea usability-wise to print this to stderr even if it is a recoverable error?

Reply:
I'm not strongly opinionated on this, but I figured it was OK since they would have compiled with '--verbose' to see this message. Right?

Reply:
That's true. I guess I'd vote for removing it, as I don't believe we currently have benign error messages being printed anywhere else in the codebase.

Reply:
Sounds good to me. I included the `st.message()` in the return error when we return no data and a non-OK status.